Plugins/Tweaks/CastAllOnHitCastSpellItemProperties.cpp

```diff
@@ -0,0 +1,76 @@
+#include "nwnx.hpp"
+
+#include "API/CAppManager.hpp"
+#include "API/CServerExoApp.hpp"
+#include "API/CServerAIMaster.hpp"
+#include "API/Constants.hpp"
+#include "API/CExoArrayList.hpp"
+#include "API/CNWSCreature.hpp"
+#include "API/CNWSItem.hpp"
+#include "API/CNWItemProperty.hpp"
+#include "API/CNWRules.hpp"
+#include "API/CNWSpellArray.hpp"
+#include "API/CNWSSpellScriptData.hpp"
+#include "API/CNWSCombatRound.hpp"
+#include "API/CNWSCombatAttackData.hpp"
+#include "API/CTwoDimArrays.hpp"
+
+namespace Tweaks {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+
+void CastAllOnHitCastSpellItemProperties() __attribute__((constructor));
+void CastAllOnHitCastSpellItemProperties()
+{
+    if (!Config::Get<bool>("CAST_ALL_ON_HIT_CAST_SPELL_ITEM_PROPERTIES", false))
+        return;
+
+    LOG_INFO("All On Hit: Cast Spell item properties will be cast on hit");
+
+    static Hooks::Hook s_ApplyOnHitCastSpellHook = Hooks::HookFunction(&CNWSCreature::ApplyOnHitCastSpell,
+        (void*)+[](CNWSCreature *pCreature, CNWSObject *pTarget, __attribute__((unused)) CNWItemProperty *pProperty, CNWSItem *pItem) -> void
```

Comment:
You can just do `CNWItemProperty*` to get rid of the unused var warning

Reply:
Oh thank you! TIL!